src/shims/unix/fd.rs

```diff
@@ -421,14 +428,14 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
         let communicate = this.machine.communicate();
 
         let bytes = this.read_bytes_ptr_strip_provenance(buf, Size::from_bytes(count))?.to_owned();
```

Comment:
Cloning the bytes should not be needed any more now that we `dup` the FD, I think?

Reply:
We needs to call `write(communicate, &bytes, this)` later, seems that there is still a conflict if `bytes` holds a 
immutable borrow to `this` :(

Reply:
Ah, right.